zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterGroup.java

```diff
@@ -50,6 +51,8 @@
   ResourcePool resourcePool;
   boolean angularRegistryPushed = false;
 
+  private RemoteInterpreterEventClient eventClient;
```

Comment:
`InterpreterGroup` is not only used in RemoteInterpreter process but also in ZeppelinServer side, too.
So having reference to eventClient in `InterpreterGroup` is not optimal i think.

I think `InterpreterContext` is better place to have a such reference, and you can inject your object in when [InterpreterContext is being created](https://github.com/apache/zeppelin/blob/master/zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java#L544).

InterpreterContext is not passed to interpreter in open(). So getting routines that getting SparkUI and update meta need to be moved from open() to interpret(). I think it's not a big problem in this use case.

Also i think it'll be better passing some wrapper object instead of eventClient directly. Because of we don't want to let interpreter directly access all eventClient's methods.